test/routes.test.js

```diff
@@ -0,0 +1,78 @@
+const index = require('..');
+const assume = require('assume');
+const thenify = require('tinythen');
+const request = require('request-promise-native');
+const StatusHandler = require('../status-handler');
+const fixtures = require('./fixtures');
+const { address } = require('./util');
+const rip = require('rip-out');
+
+
+describe('routes', function () {
+  describe('integration', function () {
+    this.timeout(6E4);
+    let status;
+    let app;
+    let spec;
+
+    before(function (done) {
+      index.start({ auth: false }, (err, instance) => {
+        app = instance;
+        status = new StatusHandler({
+          models: app.models
+        });
+        spec = status._transform(fixtures.singleEvent);
+        done(err);
+      });
+    });
+
+    after(async function () {
+      if (!app) return;
+      await app.models.drop();
+      await thenify(app, 'close');
+    });
+
+    beforeEach(async () => {
+      if (!app) return;
+      await status.event(fixtures.singleEvent);
+      await status.queued(fixtures.singleQueued);
+      await status.complete(fixtures.singleComplete);
+    });
+
+    afterEach(async () => {
+      if (!app) return;
+      const { Status, StatusHead, StatusEvent, StatusCounter } = app.models;
+      await Promise.all([
+        Status.remove(spec),
+        StatusHead.remove(spec),
+        StatusEvent.remove(spec),
+        StatusCounter.decrement(spec, 1)
+      ]);
+    });
+
+    it('/status should return status object when requested', async function () {
+      const statusObj = await request(address(app, '/status', spec));
+      assume(statusObj.complete).equals(true);
```

Comment:
Two things:

1. We should assert all of the relevant properties of the `statusObj` if there are any others. 
2. Shouldn't we be testing other states of status besides `complete`? From a test plan perspective we want to make sure all of the state transition edges (in the "clock edge" sense of the word) are tested to ensure the state machine operates correctly.

Reply:
@indexzero i can be more exhaustive, I just covered those types of assertions in the status-handler tests when fetching the models directly.